flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/source/FlinkSource.java

```diff
@@ -195,6 +205,7 @@ public FlinkInputFormat buildFormat() {
       } else {
         contextBuilder.project(FlinkSchemaUtil.convert(icebergSchema, projectedSchema));
       }
+      contextBuilder.exposeLocality(localityEnabled());
```

Comment:
I think it should be possible to override `exposeLocality` in this builder so that you can set it differently for different sources. Keeping a boolean in this builder and passing that as an override for the environment property in `localityEnabled()` should work.

Reply:
Please review whether meet expectations, thx.